dotCMS/src/main/java/com/dotcms/rendering/velocity/services/ContentletLoader.java

```diff
@@ -0,0 +1,969 @@
+              // }//http://jira.dotmarketing.net/browse/DOTCMS-3232
+            else if (field instanceof HostFolderField) {
+                if (InodeUtils.isSet(content.getFolder())) {
+                    sb.append("#set( $ConHostFolder='")
+                        .append(content.getFolder())
+                        .append("' )");
+                } else {
+                    sb.append("#set( $ConHostFolder='")
+                        .append(content.getHost())
+                        .append("' )");
+                }
+            }
+
+            else if (field instanceof CategoryField) {
+
+                // Get the Category Field
+                Category category = categoryAPI.find(field.values(), systemUser, false);
+                // Get all the Contentlets Categories
+                List<Category> selectedCategories = categoryAPI.getParents(content, systemUser, false);
+
+                // Initialize variables
+                String catInodes = "";
+                Set<Category> categoryList = new HashSet<Category>();
+                List<Category> categoryTree = categoryAPI.getAllChildren(category, systemUser, false);
+
+                if (selectedCategories.size() > 0 && categoryTree != null) {
+                    for (int k = 0; k < categoryTree.size(); k++) {
+                        Category cat = (Category) categoryTree.get(k);
+                        for (Category categ : selectedCategories) {
+                            if (categ.getInode()
+                                .equalsIgnoreCase(cat.getInode())) {
+                                categoryList.add(cat);
+                            }
+                        }
+                    }
+                }
+
+                if (categoryList.size() > 0) {
+                    StringBuilder catbuilder = new StringBuilder();
+                    Iterator<Category> it = categoryList.iterator();
+                    while (it.hasNext()) {
+                        Category cat = (Category) it.next();
+                        catbuilder.append("\"")
+                            .append(cat.getInode())
+                            .append("\"");
+                        if (it.hasNext()) {
+                            catbuilder.append(",");
+                        }
+                    }
+                    catInodes = catbuilder.toString();
+
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("FilteredCategories=$categories.filterCategoriesByUserPermissions([")
+                        .append(catInodes)
+                        .append("] ))");
+                    sb.append("#set( $")
+                        .append(field.variable())
+                        .append("Categories=$categories.fetchCategoriesInodes($")
+                        .append(field.variable())
+                        .append("FilteredCategories))");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "FilteredCategories))" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)